frontend/components/map/project-legend/component.tsx

```diff
@@ -23,28 +26,50 @@ export const ProjectLegend: FC<ProjectLegendProps> = ({ className }: ProjectLege
   const onToggleActive = useCallback(() => {
     setActive(!active);
   }, [active]);
+
   if (isLoading || isError) {
     return null;
   }
 
+  const legendVariants = {
+    open: { opacity: 1, height: 'auto' },
+    closed: { opacity: 0, height: 0 },
+  };
+
   return (
     <div
       className={cx({
-        'w-56 px-4 py-2 bg-white rounded-2xl shadow-xl text-xs flex flex-col': true,
+        'flex flex-col items-end h-full gap-px': true,
         [className]: !!className,
       })}
     >
       <button
         type="button"
-        aria-expanded={active}
         aria-controls={id}
-        className="relative flex flex-col items-start"
+        aria-expanded={active}
+        className="relative flex items-center justify-center w-8 h-8 bg-white rounded shadow-xl focus-visible:outline-green-dark"
         onClick={onToggleActive}
       >
-        <div className="font-semibold text-gray-700">
-          <FormattedMessage defaultMessage="Project categories" id="rginRA" />
-        </div>
-        {active && (
+        <Icon
+          aria-hidden={true}
+          icon={ChevronDownIcon}
+          className={cx({
+            'w-4.5 h-4.5 text-gray-600 shrink-0 transition duration-500': true,
+            '-scale-y-100': !active,
+          })}
+        />
+      </button>
```

Comment:
Accessibility-wise, everything is right with this button except… where is the label?

Here I also suggest re-using the `<Button />` component for consistent focus styles, even if the outline is slightly cut at the bottom.

```suggestion
      <Button
        theme="naked"
        size="smallest"
        aria-controls={id}
        aria-expanded={active}
        className="flex items-center justify-center w-8 h-8 bg-white rounded shadow-xl focus-visible:outline-green-dark"
        onClick={onToggleActive}
      >
        <span className="sr-only">
          <FormattedMessage defaultMessage="Legend" id="iZuO+L" />
        </span>
        <Icon
          aria-hidden={true}
          icon={ChevronDownIcon}
          className={cx({
            'w-4.5 h-4.5 text-gray-600 shrink-0 transition duration-500': true,
            '-scale-y-100': !active,
          })}
        />
      </Button>
```